cmd/controller-manager/app/controller-manager.go

```diff
@@ -177,6 +183,45 @@ func startControllers(opts *options.Options, stopChan <-chan struct{}) {
 	}
 }
 
+func setOptionsByFederationConfig(opts *options.Options) {
+	client := genericclient.NewForConfigOrDieWithUserAgent(opts.Config.KubeConfig, "federationconfig")
+
+	name := util.FederationConfigName
+	namespace := opts.Config.FederationNamespace
+	fedConfig := &corev1a1.FederationConfig{}
+	err := client.Get(context.Background(), fedConfig, namespace, name)
+	if err != nil {
+		glog.V(1).Infof("Cannot retrieve FederationConfig %q in namespace %q: %v. Command line options are used.", name, namespace, err)
+		return
+	}
+
+	glog.Infof("Setting Options with FederationConfig %q in namespace %q", name, namespace)
+
+	spec := fedConfig.Spec
+	opts.LimitedScope = spec.LimitedScope
+	opts.ClusterMonitorPeriod = spec.ControllerDuration.ClusterMonitorPeriod * time.Second
+
+	opts.Config.ClusterNamespace = spec.RegistryNamespace
+	opts.Config.ClusterAvailableDelay = spec.ControllerDuration.AvailableDelay * time.Second
+	opts.Config.ClusterUnavailableDelay = spec.ControllerDuration.UnavailableDelay * time.Second
+
+	opts.LeaderElection.ResourceLock = spec.LeaderElect.ResourceLock
+	opts.LeaderElection.RetryPeriod = spec.LeaderElect.RetryPeriod * time.Second
+	opts.LeaderElection.RenewDeadline = spec.LeaderElect.RenewDeadline * time.Second
+	opts.LeaderElection.LeaseDuration = spec.LeaderElect.LeaseDuration * time.Second
```

Comment:
Rather than assuming seconds, would it be possible to parse any given duration (e.g. `ms`, `s`, `m`, `h`)?

Reply:
I use `time.Duration` in the API to try to make it parse unit of time. However, it does not work as I expected. I will investigate and fix this.

Reply:
Good news is that this case is supported in Feb.: https://github.com/kubernetes-sigs/controller-tools/pull/138

Addressed.